prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,272 @@
+
+
+@dataclass
+class ConductorBundle:
+    """Represent a bundle of conductors (or a 'bundle' of one).
+
+    :param int n: number of conductors in bundle (can be one).
+    :param Conductor conductor: information for each conductor.
+    :param float spacing: distance between the centers of each conductor (meters).
+    :param str layout: either 'circular' (conductors are arranged in a regular polygon
+        with edge length ``spacing``) or 'flat' (conductors are arranged in a line, at
+        regular spacing ``spacing``).
+    """
+
+    n: int
+    conductor: Conductor
+    spacing: float = None  # we need to be able to ignore spacing for a single conductor
+    layout: str = "circular"
+
+    def __post_init__(self):
+        self.resistance_per_km = self.conductor.resistance_per_km / self.n
+        self.spacing_L = self.calculate_equivalent_spacing("inductance")
+        self.spacing_C = self.calculate_equivalent_spacing("capacitance")
+
+    def calculate_equivalent_spacing(self, type="inductance"):
+        if type == "inductance":
+            conductor_distance = self.conductor.gmr
+        elif type == "capacitance":
+            conductor_distance = self.conductor.radius
+        else:
+            raise ValueError("type must be either 'inductance' or 'capacitance'")
+        if self.n == 1:
+            return conductor_distance
+        elif self.n == 2:
+            return (conductor_distance * self.spacing) ** (1 / 2)
+        else:
+            if self.layout == "circular":
+                return self.calculate_equivalent_spacing_circular(conductor_distance)
+            if self.layout == "flat":
+                return self.calculate_equivalent_spacing_flat(conductor_distance)
+            raise ValueError(f"Unknown layout: {self.layout}")
+
+    def calculate_equivalent_spacing_circular(self, conductor_distance):
+        if self.n == 3:
+            return (conductor_distance * self.spacing ** 2) ** (1 / 3)
+        if self.n == 4:
+            return (conductor_distance * self.spacing ** 3 * 2 ** (1 / 2)) ** (1 / 4)
+        raise NotImplementedError("Geometry calculations not implemented for n > 4")
+
+    def calculate_equivalent_spacing_flat(self, conductor_distance):
+        if self.n == 3:
+            return (conductor_distance * 2 * self.spacing ** 2) ** (1 / 3)
+        if self.n == 4:
+            return (conductor_distance * 12 * self.spacing ** 3) ** (1 / 8)
+        raise NotImplementedError("Geometry calculations not implemented for n > 4")
+
+
+@dataclass
+class PhaseLocations:
+    """Represent the locations of each conductor bundle on a transmission tower.
+
+    :param tuple a: the (x, y) location of the single 'A' phase conductor if
+        ``circuits`` == 1, or the ((x1, y1), (x2, y2), ...) locations of the 'A' phase
+        conductors if ``circuits`` > 1. Units are meters.
+    :param tuple a: the (x, y) location of the single 'B' phase conductor if
```

Comment:
2nd and 3rd param names should be `b` and `c`

Reply:
I would consider combining the descriptions of tuple parameters in the initial docstring, before the param section. E.g. `Each of a,b,c are the (x,y) location...`

Reply:
Done.